app/frontend/src/features/BugReport.test.tsx

```diff
@@ -0,0 +1,234 @@
+>;
+
+afterEach(() => jest.restoreAllMocks);
+
+async function fillInAndSubmitBugReport(
+  subjectFieldLabel: string,
+  descriptionFieldLabel: string,
+  stepsFieldLabel: string = "",
+  resultsFieldLabel: string = ""
+) {
+  const subjectField = await screen.findByLabelText(subjectFieldLabel);
+  const descriptionField = await screen.findByLabelText(descriptionFieldLabel);
+  const stepsField = screen.queryByLabelText(stepsFieldLabel);
+  const resultsField = screen.queryByLabelText(resultsFieldLabel);
+
+  userEvent.type(subjectField, "Broken log in");
+  userEvent.type(descriptionField, "Log in is broken");
+
+  if (stepsField) {
+    userEvent.type(stepsField, "Type in user name and clicked log in");
+  }
+  if (resultsField) {
+    userEvent.type(
+      resultsField,
+      "Log in didn't work, and I expected it to work"
+    );
+  }
+
+  userEvent.click(screen.getByRole("button", { name: "Submit" }));
+}
+
+describe("BugReport", () => {
+  beforeEach(() => {
+    reportBugMock.mockResolvedValue("1");
+  });
+
+  describe("when displayed on a screen at the medium breakpoint or above", () => {
+    it("shows a button to open the bug report dialog initially", () => {
+      render(<BugReport />, { wrapper });
+
+      const reportBugButton = screen.getByRole("button", {
+        name: "Report a bug",
+      });
+      expect(reportBugButton).toBeVisible();
+      expect(reportBugButton).toHaveTextContent("Report a bug");
+    });
+  });
+
+  describe("when displayed in a small screen", () => {
+    function createMatchMedia(width: number) {
+      return (query: string) => ({
+        matches: mediaQuery.match(query, { width }),
+        addListener: jest.fn(),
+        removeListener: jest.fn(),
+      });
+    }
+
+    beforeEach(() => {
+      // @ts-ignore JSDom doesn't enforce this to be readonly, so less verbose to do this than
+      // using Object.defineProperty(...)
+      window.innerWidth = 959;
+      // @ts-ignore
+      window.matchMedia = createMatchMedia(window.innerWidth);
+    });
+
+    afterEach(() => {
+      // @ts-ignore reset back to default JSDom window width
+      window.innerWidth = 1024;
+      // @ts-ignore unset to prevent this from interferring tests below
+      window.matchMedia = undefined;
+    });
+
+    it("shows a button with only the bug report icon", () => {
+      render(<BugReport />, { wrapper });
+      const reportBugButton = screen.getByRole("button", {
+        name: "Report a bug",
+      });
+      expect(reportBugButton).toBeVisible();
+      expect(reportBugButton).not.toHaveTextContent("Report a bug");
+    });
+  });
+
+  describe('when the "report a bug" button is clicked', () => {
+    const subjectFieldLabel = "Brief description of the bug";
+    const descriptionFieldLabel = "What's the problem?";
+    const stepsFieldLabel = "What did you do to trigger the bug?";
+    const resultsFieldLabel =
+      "What happened? What did you expect should have happened?";
```

Comment:
This is an interesting one... the author of React Testing Library [advise to still assert by text rather than using test IDs](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-querying-by-text) even in a multi-lingual app so we can ensure at least one language is being translated correctly.

But yeah, definitely something worth re-visiting if this becomes a pain as I don't feel as strongly as him on this. The Twitter thread he linked to on that paragraph also suggests some good solution on selecting by the translation key text, run the translation function on it (which we can probably set to English in the tests) and assert the result of that instead.

Reply:
Yeah I just don't like the idea of having to open multiple files to make a label change. But with translation keys it makes sense.